Allow templating of `host` and `advertisedHost` fields (#10605)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.kafka.listener.GenericKafkaListenerConfiguration.adoc

# ...
----

> Instead of specifying the `host` field for every broker, you can also use an `hostTemplate` to generate them automatically.

Reviewer: ```suggestion
Instead of specifying the `host` property for every broker, you can also use a `hostTemplate` to generate them automatically.
```

The `hostTemplate` supports the following variables:

> * The `{nodeId}` variable will be replaced with the ID of the Kafka node to which the template is applied.
> * The `{nodePodName}` variable will be replaced with the Kubernetes pod name for the Kafka node where the template is applied.

Reviewer: ```suggestion
* The `{nodeId}` variable is replaced with the ID of the Kafka node to which the template is applied.
* The `{nodePodName}` variable is replaced with the Kubernetes pod name for the Kafka node where the template is applied.
```

> The `hostTemplate` field is applied only to the per-broker `host` fields.
> The bootstrap `host` field has to be always specified.

Reviewer: ```suggestion
The `hostTemplate` property applies only to per-broker values.
The bootstrap `host` property must always be specified.
```

> .Example `ingress` listener with the `hostTemplate`

Reviewer: ```suggestion
.Example `ingress` listener with `hostTemplate` configuration
```
